qiskit/circuit/library/generalized_gates/linear_function.py

```diff
@@ -171,22 +254,47 @@ def permutation_pattern(self):
         locs = np.where(linear == 1)
         return locs[1]
 
+    def extend_with_identity(self, num_qubits: int, positions: list[int]) -> LinearFunction:
+        """Extend linear function to a linear function over nq qubits,
+        with identities on other subsystems.
 
-def _linear_quantum_circuit_to_mat(qc: QuantumCircuit):
-    """This creates a n x n matrix corresponding to the given linear quantum circuit."""
-    nq = qc.num_qubits
-    mat = np.eye(nq, nq, dtype=bool)
-
-    for instruction in qc.data:
-        if instruction.operation.name == "cx":
-            cb = qc.find_bit(instruction.qubits[0]).index
-            tb = qc.find_bit(instruction.qubits[1]).index
-            mat[tb, :] = (mat[tb, :]) ^ (mat[cb, :])
-        elif instruction.operation.name == "swap":
-            cb = qc.find_bit(instruction.qubits[0]).index
-            tb = qc.find_bit(instruction.qubits[1]).index
-            mat[[cb, tb]] = mat[[tb, cb]]
-        else:
-            raise CircuitError("A linear quantum circuit can include only CX and SWAP gates.")
+        Args:
+            num_qubits: number of qubits of the extended function.
 
-    return mat
+            positions: describes the positions of original qubits in the extended
+                function's qubits.
+
+        Returns:
+            LinearFunction: extended linear function.
+        """
+        extended_mat = np.eye(num_qubits, dtype=bool)
+
+        for i, pos in enumerate(positions):
+            extended_mat[positions, pos] = self.linear[:, i]
+
+        return LinearFunction(extended_mat)
+
+    def mat_str(self):
+        """Return string representation of the linear function
+        viewed as a matrix with 0/1 entries.
+        """
+        return str(self.linear.astype(int))
+
+    def function_str(self):
+        """Return string representation of the linear function
+        viewed as a linear transformation.
+        """
+        out = "("
+        mat = self.linear
```

Comment:
I might be missing something but where do we assign `self.linear`, I didn't see it in `__init__()`?

Reply:
Initially `linear` was indeed a field in `LinearFunction`, but at some point it was moved into `params`, so now `linear` is a property     
```
@property
def linear(self):
    """Returns the n x n matrix representing this linear function."""
    return self.params[0]
```

Reply:
Ah yep, that's what I was missing thanks.